### DISCUSSION TITLE
python-OpcUA to convert OPC DA into OPC UA

### DISCUSSION THREAD

**Discussion by Adarsh-Pansari** [FreeOpcUa/python-opcua#1250] [Q&A] [answered]
Hello All,
Is it possible to use the library to convert the OPC DA to OPC UA.

**Comment by swamper123** [comment, marked as answer]
Hey ho,

this library doesn't support OPC DA, just OPC UA. You have to build a workaround yourself for that.

**Reply by AndreasHeine** [reply]
in order to convert from "OPC DA" to "OPC UA" you need to convert COM/DCOM to TCP/IP (OPC UA Binary Protocol)
this lib can only provide the opc ua part for the com/dcom part you need a different lib...
but there are gateways which already do that stuff for you -> i would assume it is cheaper to buy it in order to code it yourself
